client/state/sites/features/reducer.js

```diff
@@ -25,6 +29,26 @@ function updateSiteState( state, siteId, attributes ) {
 	} );
 }
 
+/**
+ * Given an object of features keyed by siteId, updates all features at once.
+ *
+ * @param {object} state The current features state
+ * @param {object} features An object containing feature arrays keyed by siteId
+ * @returns The new state
+ */
+function updateBulkFeatures( state, features ) {
```

Comment:
This allows redux to be updated only once when the features are retrieved for all sites, thereby reducing the number of render passes.